src/main/scala/gitbucket/core/service/PullRequestService.scala

```diff
@@ -293,6 +302,66 @@ trait PullRequestService {
     }
   }
 
+  def updatePullRequestsByApi(
+    repository: RepositoryInfo,
+    issueId: Int,
+    loginAccount: Account,
+    settings: SystemSettings,
+    title: Option[String],
+    body: Option[String],
+    state: Option[String],
+    base: Option[String]
+  )(
+    implicit s: Session,
+    c: JsonFormat.Context
+  ): Unit = {
+    getPullRequest(repository.owner, repository.name, issueId) match {
+      case Some(x) =>
+        if (Repositories.filter(_.byRepository(x._2.userName, x._2.repositoryName)).exists.run) {
+          // Update base branch
+          base match {
+            case Some(_base) =>
+              Using.resource(Git.open(getRepositoryDir(repository.owner, repository.name))) { git =>
+                getBranches(git, repository.repository.defaultBranch, origin = true).find(_.name == _base) match {
```

Comment:
In this case, using `foreach` instead of `match` would be better so that we can get rid of `case _ =>`.